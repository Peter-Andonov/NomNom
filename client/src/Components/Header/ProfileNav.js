import React, { useState, useContext } from 'react';
import styled from 'styled-components';
import UserContext from '../../Context';
import TextItem from './TextItem';


const Container = styled.div`
    margin: 1rem;
    color: white;
    text-decoration: none;
    display: flex;
    flex-direction: row;
    align-items: center;
    &:hover {
        cursor: pointer;
    }
`;

const Avatar = styled.img`
    margin-right: 1rem;
    width: 4rem;
    height: 4rem;
    border-radius: 50%;
    object-fit: cover;
`;


const ProfileNav = (props) => {

    const [open, setOpen] = useState(false);

    const toggleOpen = () => {
        setOpen(!open)
    };

    const userContext = useContext(UserContext);
    const displayName = userContext.user.firstName ? `${userContext.user.firstName} ${userContext.user.lastName}` : userContext.user.email;
    const displayPicture = userContext.user.profilePicUrl || 'https://res.cloudinary.com/nomnomapp/image/upload/v1596641498/Images/x5svb4l6kwcyiinazfqt.png';
    
    return (
        <Container onClick={toggleOpen}>
            <Avatar src={displayPicture} alt='Avatar' />
            <TextItem label={displayName} />
            {open && props.children}
        </Container>
    );

};


export default ProfileNav;