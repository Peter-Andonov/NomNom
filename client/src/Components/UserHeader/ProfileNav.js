import React, { useState, useContext } from 'react';
import styled from 'styled-components';
import UserContext from '../../Context';


const Container = styled.div`
    margin: 1rem;
    color: white;
    text-decoration: none;
    display: flex;
    flex-direction: row;
    align-items: center;
    &:hover {
        cursor: pointer;
    }
`;

const Avatar = styled.img`
    height: 50%;
    max-width: 4rem;
    max-height: 4rem;
    border-radius: 50%;
    object-fit: cover;
`;

const Text = styled.span`
    margin-left: 1rem;
`;


const ProfileNav = (props) => {

    const [open, setOpen] = useState(false);

    const toggleOpen = () => {
        setOpen(!open)
    }

    const userContext = useContext(UserContext);
    const displayName = userContext.user.firstName ? `${userContext.user.firstName} ${userContext.user.lastName}` : userContext.user.email;
    const displayPicture = userContext.user.profilePicUrl || 'https://res.cloudinary.com/nomnomapp/image/upload/v1596641498/Images/x5svb4l6kwcyiinazfqt.png';
    
    return (
        <Container onClick={toggleOpen}>
            <Avatar src={displayPicture} alt='Avatar' />
            <Text>{displayName}</Text>
            {open && props.children}
        </Container>
    );

};


export default ProfileNav;