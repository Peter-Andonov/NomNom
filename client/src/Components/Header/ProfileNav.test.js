import React from 'react'
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import TestSetup from '../../Utils/TestUtils/test-utils';


import ProfileNav from './ProfileNav.js';

describe('ProfileNav component', () => {
    it("renders default avatar picture and email", () => {
        const { asFragment } = render(
            <TestSetup value={{
                loggedIn: false,
                user: {
                    _id: '1234',
                    email: "john.doe@gmail.com",
                    role: 'user',
                    firstName: undefined,
                    lastName: undefined,
                    profilePicUrl: undefined,
                    favouriteRecipes: []
                }
            }}>
                <ProfileNav />
            </TestSetup>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders user uploaded profile picture", () => {
        const { asFragment } = render(
            <TestSetup value={{
                loggedIn: false,
                user: {
                    _id: '1234',
                    email: "john.doe@gmail.com",
                    role: 'user',
                    firstName: undefined,
                    lastName: undefined,
                    profilePicUrl: 'https://i.picsum.photos/id/1005/5760/3840.jpg?hmac=2acSJCOwz9q_dKtDZdSB-OIK1HUcwBeXco_RMMTUgfY',
                    favouriteRecipes: []
                }
            }}>
                <ProfileNav />
            </TestSetup>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders default profile pic and user names instead of email", () => {
        const { asFragment } = render(
            <TestSetup value={{
                loggedIn: false,
                user: {
                    _id: '1234',
                    email: "john.doe@gmail.com",
                    role: undefined,
                    firstName: 'John',
                    lastName: 'Doe',
                    profilePicUrl: undefined,
                    favouriteRecipes: []
                }
            }}>
                <ProfileNav />
            </TestSetup>);
        expect(asFragment()).toMatchSnapshot();
    });
});