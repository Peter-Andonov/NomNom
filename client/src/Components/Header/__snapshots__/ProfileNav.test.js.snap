// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileNav component renders default avatar picture and email 1`] = `
<DocumentFragment>
  <div
    class="sc-AxirZ hVzgMg"
  >
    <img
      alt="Avatar"
      class="sc-AxiKw fiBQpN"
      src="https://res.cloudinary.com/nomnomapp/image/upload/v1596641498/Images/x5svb4l6kwcyiinazfqt.png"
    />
    <div
      class="sc-AxjAm cxNAeo"
    >
      john.doe@gmail.com
    </div>
  </div>
</DocumentFragment>
`;

exports[`ProfileNav component renders default profile pic and user names instead of email 1`] = `
<DocumentFragment>
  <div
    class="sc-AxirZ hVzgMg"
  >
    <img
      alt="Avatar"
      class="sc-AxiKw fiBQpN"
      src="https://res.cloudinary.com/nomnomapp/image/upload/v1596641498/Images/x5svb4l6kwcyiinazfqt.png"
    />
    <div
      class="sc-AxjAm cxNAeo"
    >
      John Doe
    </div>
  </div>
</DocumentFragment>
`;

exports[`ProfileNav component renders user uploaded profile picture 1`] = `
<DocumentFragment>
  <div
    class="sc-AxirZ hVzgMg"
  >
    <img
      alt="Avatar"
      class="sc-AxiKw fiBQpN"
      src="https://i.picsum.photos/id/1005/5760/3840.jpg?hmac=2acSJCOwz9q_dKtDZdSB-OIK1HUcwBeXco_RMMTUgfY"
    />
    <div
      class="sc-AxjAm cxNAeo"
    >
      john.doe@gmail.com
    </div>
  </div>
</DocumentFragment>
`;
